Add a pull request template (#407)
substrate-developer-hub/substrate-developer-hub.github.io: CONTRIBUTING.md

* [Documentation Style](#documentation-style)

## PR Checklist

> - [ ] Is the audience and objective of the document clear from the first paragraph? E.g. a document

Reviewer: I realized after I wrote this that it could also be clear just from its location on the website. Not everything should have to say "for a developer" if it's in the "Developer Docs" section. Audience/objective should be clear from context.
```suggestion
- [ ] Are the audience and objective of the document clear? E.g. a document
```

for developers that should teach them about transaction fees.
- [ ] Is the writing:
  - **Clear**: No jargon.
  - **Precise**: No ambiguous meanings.
  - **Concise**: Free of superfluous detail.
- [ ] Does it follow our [style guide](#documentation-style)?
- [ ] If this is a new page, does the PR include the appropriate infrastructure, e.g. adding the
page to a sidebar?
- [ ] Build the page (`$ cd website && yarn start`). Does it render properly? E.g. no funny lists
or formatting.

> - [ ] Do links to rustdocs or devhub articles rather than code?

Reviewer: ```suggestion
- [ ] Do links go to rustdocs or devhub articles rather than code?
```
